projects/frontegg-app/src/lib/frontegg-entitlements.service.ts

```diff
@@ -39,57 +36,58 @@ export class FronteggEntitlementsService {
    * No need for deep equal because we already check it internally
    * @param authState
    */
-  private updateEntitlementsStateIfNeeded(authState: AuthState): void {
-    const entitlementsState = authState.user?.entitlements as any;
-    if (this.entitlementsStateSubject.value === entitlementsState) {
+  private updateUserStateIfNeeded(authState: AuthState): void {
+    const entitlementsState = authState.user as User;
+    if (this.userStateSubject.value === entitlementsState) {
       return;
     }
 
-    this.entitlementsStateSubject.next(entitlementsState);
+    this.userStateSubject.next(entitlementsState);
```

Comment:
will this effect the other user subject in auth service? why do we have duplicated userSubject?

Reply:
it won't affect other subjects. We use a separate one because of the types issue (admin box testing shit)